op/k8s/block_device_move.go

```diff
@@ -0,0 +1,188 @@
+package k8s
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/cybozu-go/cke"
+	"github.com/cybozu-go/cke/op"
+	"github.com/cybozu-go/log"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type blockDeviceMoveOp struct {
+	nodes []*cke.Node
+	step  int
+}
+
+// BlockDeviceMoveOp returns an Operator to restart kubelet
+func BlockDeviceMoveOp(nodes []*cke.Node) cke.Operator {
+	return &blockDeviceMoveOp{nodes: nodes}
+}
+
+func (o *blockDeviceMoveOp) Name() string {
+	return "block-device-move"
+}
+
+func (o *blockDeviceMoveOp) Targets() []string {
+	ips := make([]string, len(o.nodes))
+	for i, n := range o.nodes {
+		ips[i] = n.Address
+	}
+	return ips
+}
+
+func (o *blockDeviceMoveOp) NextCommand() cke.Commander {
+	switch o.step {
+	case 0:
+		o.step++
+		return moveBlockDeviceFor17(o.nodes)
+	default:
+		return nil
+	}
+}
+
+type moveBlockDeviceFor17Command struct {
+	nodes []*cke.Node
+}
+
+// moveBlockDeviceFor17 move raw block device files.
+// This command is used for upgrading to k8s 1.17
+func moveBlockDeviceFor17(nodes []*cke.Node) cke.Commander {
+	return moveBlockDeviceFor17Command{nodes}
+}
+
+func (c moveBlockDeviceFor17Command) Run(ctx context.Context, inf cke.Infrastructure, _ string) error {
+	for _, n := range c.nodes {
+		clientset, err := inf.K8sClient(ctx, n)
+		if err != nil {
+			return err
+		}
+
+		agent := inf.Agent(n.Address)
+		if agent == nil {
+			return errors.New("unable to prepare agent for " + n.Nodename())
+		}
+
+		stdout, stderr, err := agent.Run(fmt.Sprintf("find %s -type b", op.CSIBlockDevicePublishDirectory))
```

Comment:
This function would run too slowly when there are many nodes.
Run remote commands in parallel using `well.NewEnvironment` just like other commands do.